### PR TITLE
Cancel appointment by deleting DRS booking

### PR DESCRIPTION
**Pull request by Bhavesh-Patel**

### CODE REVIEW

**Comment by Bhavesh-Patel** [comment]
[Create backend API repair appointment cancellation endpoint](https://trello.com/c/97sjt8Mr/242-create-backend-api-repair-appointment-cancellation-endpoint)

File: HousingRepairsSchedulingApi/Services/Drs/DrsService.cs [hunk 1]
```diff
@@ -184,6 +184,27 @@ public async Task<order> SelectOrder(string bookingReference)
             return drsOrder;
         }
 
+        public async Task DeleteBooking(string bookingReference, int bookingId)
+        {
+            Guard.Against.NullOrWhiteSpace(bookingReference, nameof(bookingReference));
+
```
Comment (markWarr) [inline HousingRepairsSchedulingApi/Services/Drs/DrsService.cs:190]: Do we need any validation on the bookingId, i.e. > 0?

Comment (Bhavesh-Patel) [reply HousingRepairsSchedulingApi/Services/Drs/DrsService.cs:190]: Although it's unlikely DRS booking Ids allow negative values, without knowing for certain, I decided to omit validation.

Validation could be added quite easily though.
Should we make an assumption they are positive values and add validation?

Comment (markWarr) [reply HousingRepairsSchedulingApi/Services/Drs/DrsService.cs:190]: Perhaps it's overkill to check for negative values, as you say we can't be sure that booking ids can't be negative.

**Review by markWarr — DISMISSED** [review]

**Review by markWarr — APPROVED** [review]

**Pull request merged**